src/js/models/menuitem.js

```diff
@@ -86,21 +82,43 @@ define(function (require, exports, module) {
         commandKind: null,
 
         /**
-         * @type {{modifiers: number=, keyCode: number}}
+         * @type {?Shortcut}
          */
         shortcut: null,
 
         /**
          * @type {boolean}
          */
-        enabled: null,
+        enabled: true,
 
         /**
-         * @type {number}
+         * @type {boolean}
          */
-        checked: null
+        checked: false
     });
 
+    Object.defineProperties(MenuItem.prototype, object.cachedGetSpecs({
+        /**
+         * Maps from itemID to submenus, sans separators
+         * @type {Immutable.Map.<string, MenuItem>} 
+         */
+        submenuMap: function () {
+            if (!this.submenu) {
+                return null;
+            }
+
+            return Immutable.Map(this.submenu
+                .toSeq()
+                .filterNot(function (entry) {
+                    return entry.type === "separator";
+                })
+                .map(function (entry) {
+                    return [entry.itemID, entry];
+                })
+                .toList());
```

Comment:
But does Map constructor accept a sequence?  Mainly I'm responding to this comment so that I'll start getting notifications of future comments.